sopa/patches/infer.py

```diff
@@ -72,38 +74,44 @@ def _get_extraction_parameters(
     resize_factor = slide_metadata["level_downsamples"][level] / downsample
     patch_width = int(patch_width * downsample)
 
-    return level, resize_factor, patch_width, True
+    return level, resize_factor, patch_width, downsample, True
 
 
-class Embedder:
+class Inference:
     def __init__(
         self,
         image: MultiscaleSpatialImage | SpatialImage,
-        model_name: str,
+        model: Callable | str,
         patch_width: int,
         level: int | None = 0,
         magnification: int | None = None,
         device: str = "cpu",
     ):
         self.image = image
-        self.model_name = model_name
+
+        if isinstance(model, str):
+            assert hasattr(
+                models, model
+            ), f"'{model}' is not a valid model name under `sopa.embedding.models`. Valid names are: {', '.join(models.__all__)}"
+            self.model_str = model
```

Comment:
I added the `self.model_str` to get the value also in case the input is a Callable and use it at the end of the function to name the key for the inferred image